calabash/Classes/FranklyServer/LPRouter.m

```diff
@@ -206,11 +206,21 @@ - (BOOL) supportsMethod:(NSString *) method atPath:(NSString *) path {
   if ([[NSThread currentThread] isMainThread]) {
     return [self httpResponseOnMainThreadForMethod:method URI:path];
   } else {
-    __weak typeof(self) wself = self;
-    __block NSObject<LPHTTPResponse> *result = nil;
-    dispatch_sync(dispatch_get_main_queue(), ^{
-      result = [wself httpResponseOnMainThreadForMethod:method URI:path];
-    });
+    // Use performSelectorOnMainThread instead of dispatch_sync because using a dispatch method prevents
```

Comment:
But if we remove the comment it's not going to be clear to a future maintainer of this code why we need to use performSelectorOnMainThread instead of dispatch_sync here. The danger is that the code could be reverted back to dispatch_sync as its not going to be immediately obvious to another developer why performSelectorOnMainThread on main thread is necessary here. Perhaps we can amend the comment to make it more concise? I look forward to your suggestions.